scripts/regions.rb

```diff
@@ -7,60 +7,46 @@
 require 'parallel'
 
 data_dir = './_data'
-websites = JSON.parse(File.read("#{data_dir}/all.json"))
-regions = YAML.load_file("#{data_dir}/regions.yml")
-tmp_dir = '/tmp'
-regions.insert(0, { 'id' => 'int', 'name' => 'global' })
-git_dir = Dir.glob('.git')
-FileUtils.cp_r(git_dir, "#{tmp_dir}/") unless File.exist?("#{tmp_dir}/.git")
+categories = JSON.parse(File.read("#{data_dir}/categories.json"))
+regions = YAML.load_file("#{data_dir}/regions.yml").map { |r| [r['id'], r] }.to_h
+regions['int'] = { 'name' => 'Global' }
+tmp_dir = `mktemp -d`.strip
+
+FileUtils.cp_r(Dir.glob('.git'), "#{tmp_dir}/") unless File.exist?("#{tmp_dir}/.git")
 
 # Region loop
-# rubocop:disable Metrics/BlockLength
-# rubocop:disable Layout/LineLength
-Parallel.each(-> { regions.pop || Parallel::Stop }) do |region|
-  dest_dir = "#{tmp_dir}/#{region['id']}"
+Parallel.map(regions.keys) do |key|
+  region = regions[key]
+  dest_dir = "#{tmp_dir}/#{key}"
   Dir.mkdir(dest_dir) unless File.exist?(dest_dir)
-  files = %w[index.html _includes _layouts _data]
-  FileUtils.cp_r(files, dest_dir)
-
+  FileUtils.cp_r(%w[index.html _includes _layouts _data], dest_dir)
   File.open("#{dest_dir}/_config_region.yml", 'w') do |file|
-    file.write("title: 2FA Directory (#{region['name']})") unless region['id'].eql?('int')
+    file.write("title: 2FA Directory (#{region['name']})") unless key.eql?('int')
   end
 
   all = {}
-  used_categories = {}
+  used_categories = []
 
   # Website loop
-  websites.each do |name, website|
-    unless website['regions'].nil?
-      site_regions = website['regions'].reject { |r| r.start_with?('-') }
-      site_excluded_regions = website['regions'].select { |r| r.start_with?('-') }.map! { |region_code| region_code.tr('-', '') }
-    end
-
-    unless website['regions'].nil? || site_regions.empty? || site_regions.include?(region['id']) || region['id'].eql?('int')
-      next
-    end
-
-    next if !site_excluded_regions.nil? && site_excluded_regions.include?(region['id'])
+  JSON.parse(File.read("#{data_dir}/all.json")).each do |name, website|
+    # Ignore if current region is excluded
+    next if website['regions']&.select { |r| r.start_with?('-') }&.map! { |rc| rc.tr('-', '') }&.include?(region['id'])
+    # Ignore unless regions element includes current region or current region is int
+    next unless website['regions']&.reject { |r| r.start_with?('-') }&.include?(key) || key.eql?('int')
```

Comment:
```suggestion
    next if website['regions']&.select { |r| r.start_with?('-') }&.map! { |rc| rc.tr('-', '') }&.include?(key)
    # Ignore unless site includes current region, current region is int, site has no regions or site is not excluded in current region
    regions = website['regions']&.reject { |r| r.start_with?('-') }
    next unless regions&.include?(key) || key.eql?('int') || website['regions'].nil? || regions.empty?
```
Current code causes sites with no regions to not be displayed on regional sites and sites with excluded regions to not be displayed on any sites.